HomePage/index.html

```diff
@@ -0,0 +1,141 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>X Mart | Home</title>
+    <link rel="stylesheet" href="style.css" />
+    <script src="./index.js"></script>
+    <script src="./"></script>
+    <link
+      rel="stylesheet"
+      href="https://fonts.googleapis.com/icon?family=Material+Icons+Outlined"
+    />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@300&family=Quicksand:wght@500&display=swap"
+      rel="stylesheet"
+    />
+  </head>
+  <body>
+    <header>
+      <div class="head">
+        <h1>X Mart</h1>
+      </div>
+      <div class="input input-search">
+        <i><span class="material-icons-outlined"> search </span></i>
+        <input class="search-input" type="text" placeholder="Search" />
+      </div>
+      <nav class="page-nav">
+        <ul>
+          <div class="icon-with-badge">
+            <a href="../cart/cart.html"
+              ><i
+                ><span class="material-icons-outlined">shopping_cart</span></i
+              ></a
+            >
+
+            <div class="badge badge-icon">3</div>
+          </div>
+          <div class="icon-with-badge">
+            <a href="../wishlist/wishlist.html"
+              ><i
+                ><span
+                  class="material-icons-outlined"
+                  style="color: rgb(226, 17, 17)"
+                  >favorite</span
+                >
+              </i></a
+            >
+            <div class="badge badge-icon">3</div>
+          </div>
+          <a href="../index.html"
+            ><img class="avatar-smallest" src="https://picsum.photos/100"
+          /></a>
+        </ul>
+      </nav>
+    </header>
+    <section class="hero-img-container">
+      <img class="hero-img" src="./images/HomePageImage.jpg" />
+      <div class="hero-cta-container">
+        <h2 class="hero-title">
+          Your One Spot destination to buy all treding clothes out there
+        </h2>
+        <div class="hero-cta" onclick="handleCategoryClick()">Shop Now</div>
```

Comment:
It's best practice to keep logic and structure separated. You could use a script tag or an external file for the functionality.